website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+    self.index = index;  // For breadcrumbs to cut off when clicking parent level
+    self.generateLink = function () {
+        if (self.type === 'collection'){
+                return $osf.apiV2Url(self.data.path, {
+                        query : self.data.query
+                    }
+                );
+        }
+        else if (self.type === 'tag') {
+            return $osf.apiV2Url('nodes/', { query : {'filter[tags]' : self.data.tag , 'related_counts' : true, 'embed' : 'contributors'}});
+        }
+        else if (self.type === 'name') {
+            return $osf.apiV2Url('users/' + self.data.id + '/nodes/', { query : {'related_counts' : true, 'embed' : 'contributors' }});
+        }
+        else if (self.type === 'node') {
+            return $osf.apiV2Url('nodes/' + self.data.uid + '/children/', { query : { 'related_counts' : true, 'embed' : 'contributors' }});
+        }
+        // If nothing
+        throw new Error('Link could not be generated from linkObject data');
+    };
+    self.link = self.generateLink();
+};
+
+
+function _makeTree (flatData) {
+    var root = {id:0, children: [], data : {} };
+    var node_list = { 0 : root};
+    var parentID;
+    for (var i = 0; i < flatData.length; i++) {
+        var n = _formatDataforPO(flatData[i]);
+        if (!node_list[n.id]) { // If this node is not in the object list, add it
+            node_list[n.id] = n;
+            node_list[n.id].children = [];
+        } else { // If this node is in object list it's likely because it was created as a parent so fill out the rest of the information
+            n.children = node_list[n.id].children;
+            node_list[n.id] = n;
+        }
+
+        if (n.relationships.parent){
+            parentID = n.relationships.parent.links.related.href.split('/')[5]; // Find where parent id string can be extracted
+        } else {
+            parentID = null;
+        }
+        if(parentID && !n.attributes.registration ) {
+            if(!node_list[parentID]){
+                node_list[parentID] = { children : [] };
+            }
+            node_list[parentID].children.push(node_list[n.id]);
+
+        } else {
+            node_list[0].children.push(node_list[n.id]);
+        }
+    }
+    return root.children;
+}
+
+/**
+ * Returns the object to send to the API to end a node_link to collection
+ * @param id {String} unique id of the node like 'ez8f3'
+ * @returns {{data: {type: string, relationships: {nodes: {data: {type: string, id: *}}}}}}
+ */
+function buildCollectionNodeData (id) {
+    return {
+        'data' : {
+            'type': 'node_links',
+            'relationships': {
+                'nodes': {
+                    'data': {
+                        'type': 'nodes',
+                        'id': id
+                    }
+                }
+            }
+        }
+    };
+}
+
+/**
+ * Initialize File Browser. Prepeares an option object within FileBrowser
+ * @constructor
+ */
+var FileBrowser = {
```

Comment:
Why "FileBrowser"? The implementation of this seems tied to ProjectOrganizer, so I would expect this to be named something like "Dashboard".